test/e2e/provisioning/testdata/machinedeployment-baremetal-tinkerbell.yaml

```diff
@@ -25,6 +25,14 @@ spec:
           cloudProvider: "baremetal"
           cloudProviderSpec:
             driver: "tinkerbell"
+            metadataClient:
+              config:
+                endpoint: ""
+                authConfig:
+                  authMethod: ""
+                  username: ""
```

Comment:
I would rather use a secret for client credentials

Reply:
So done in this commit https://github.com/kubermatic/machine-controller/pull/956/commits/c7e104ea8259550435cfc9a43215ceb1587863cf